src/components/editor/MonacoEditor.vue

```diff
@@ -24,36 +28,51 @@ const props = defineProps({
     type: String,
     required: true,
   },
-  fileInput: {
+  file: {
     type: Object,
     require: true,
   },
 });
 
 const container = ref(null);
 let editor;
+let checkoutSubscription;
+let addRemoteSubscription;
+let pullSubscription;
+let updateFileContentSubsciption;
 
 /**
  * Update file content on fs and emit an event.
  */
 async function updateFile() {
-  const file = { ...props.fileInput, content: editor.getValue() };
-  file.path = file.id;
+  const file = {
+    ...props.file,
+    path: props.file.id,
+    content: editor.getValue(),
+  };
+
   await writeProjectFile(props.projectName, file);
-  FileEvent.UpdateFileEvent.next(props.fileInput.id);
+  FileEvent.UpdateEditorContentEvent.next(file.id);
+}
+
+function getFileContent() {
+  return readProjectFile(props.projectName, { path: props.file.id })
+    .then((fileInput) => fileInput.content);
 }
 
 /**
  * Setup monaco editor.
  */
-function createEditor() {
+async function createEditor() {
+  // TODO : add loader
```

Comment:
todo,forget something ?

Reply:
I think when we talked about adding a loader, we said to do it later (out of scope for this PR)